hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/ITTestHoodieDataSource.java

```diff
@@ -480,7 +480,259 @@ void testStreamReadWithDeletes() throws Exception {
     final String expected = "[+I(+I[Danny, 24]), +I(+I[Stephen, 34])]";
     assertRowsEquals(result, expected, true);
   }
+  //scene01 compaction off
+  @Test
```

Comment:
let's use `@ParameterizedTest` to merge the tests, seems we only need two tests, string ordering fields with compaction off/on, decimal ordering fields with compaction off/on.